### PR TITLE
Merge bitcoin#19733, bitcoin#19734

### PR DESCRIPTION
**Pull request by vijaydasmp**
Merge bitcoin#19733: Move comment about BaseIndex::DB from TxIndex::DB …
Merge bitcoin#19734: Move CDiskTxPos to its own file

### CODE REVIEW

**Comment by vijaydasmp** [comment]
@PastaPastaPasta  @UdjinM6 
Getting system failure messages from CI system, somebody please check the CI system 

> There has been a runner system failure, please try again

**Comment by PastaPastaPasta** [comment]
> Hello @PastaPastaPasta @UdjinM6
> Getting system failure messages from CI system, somebody please check the CI system
> 
> > There has been a runner system failure, please try again

Have restarted those runners

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by vijaydasmp** [comment]
Thanks  @UdjinM6 , will work on those

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -286,28 +286,26 @@ UniValue deriveaddresses(const JSONRPCRequest& request)
 {
     if (request.fHelp || request.params.empty() || request.params.size() > 3) {
         throw std::runtime_error(
-            RPCHelpMan{"deriveaddresses",
-            {"\nDerives one or more addresses corresponding to an output descriptor.\n"
+            "deriveaddresses"
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply src/rpc/misc.cpp:289]: Thanks @UdjinM6 , I observed that doc/descriptors.md this file was not present , However there is 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/vijay/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/vijay/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<style>

</style>
</head>

<body link="#1155CC" vlink="#1155CC">



  |   | 0f681ceca | Add descriptor reference   documentation |   | FALSE
-- | -- | -- | -- | -- | --




</body>

</html>
an entry in the Backport sheet for 0.17 version which wasn't done as it was not marked as "Merged Done" there , So probably I have to pull this change to fix doc/descriptors.md changes

File: src/rpc/client.cpp [hunk 2]
```diff
@@ -96,8 +96,7 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "sendmany", 6, "use_is" },
     { "sendmany", 7, "use_cj" },
     { "sendmany", 8, "conf_target" },
-    { "deriveaddresses", 1, "begin" },
-    { "deriveaddresses", 2, "end" },
+    { "deriveaddresses", 1, "range" },
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply src/rpc/client.cpp:99]: Hello @UdjinM6 I tried to implement 15497 , It required Merge #14726 which is staged with 
Vlad_Staged (staged to Vladimir (dzutte-cpp) local repo) which is waiting on 
Merge#14023 which is staged with you @UdjinM6 , Now we are in Blocked chain :)

File: src/rpc/misc.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply src/rpc/misc.cpp:289]: I observed that doc/descriptors.md this file was not present,  There was a change mentioned Backport sheet for 0.17 version in 0f681ce (Add descriptor reference documentation) , however this changed was not merged , thats why this file was not present	, will pull this changes

**Comment by vijaydasmp** [comment]
In descriptor_test.cpp
```
void Check(const std::string& prv, const std::string& pub, int flags, const std::vector<std::vector<std::string>>& scripts, const std::set<std::vector<uint32_t>>& paths = ONLY_EMPTY)
{
    FlatSigningProvider keys_priv, keys_pub;
    std::set<std::vector<uint32_t>> left_paths = paths;

    // Check that parsing succeeds.
    std::unique_ptr<Descriptor> parse_priv = Parse(MaybeUseHInsteadOfApostrophy(prv), keys_priv);
    std::unique_ptr<Descriptor> parse_pub = Parse(MaybeUseHInsteadOfApostrophy(pub), keys_pub);
    BOOST_CHECK(parse_priv);
    BOOST_CHECK(parse_pub);  <=== This boost check is failing ,
```
trying to understand why

**Comment by vijaydasmp** [comment]
(Dropped the commits which are dependent on other commits for now) ,
Raising PR to Merge 19733 and 19734

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**